### DISCUSSION TITLE
compatibility of c++ library and  go-capnp

### DISCUSSION THREAD

**Discussion by angelskieglazki** [capnproto/capnproto#1992] [Q&A] [answered]
Hi all.
Does anyone know if there is any compatible version of the Go capnproto library with C++ version 1.0.2? I want to use a library for serialization and deserialization and I have one server written in C++, and many services written in Go.

regards, max 🤗

**Comment by kentonv** [comment, marked as answer]
All versions Cap'n Proto are binary compatible with each other -- the binary encoding has never changed since the first release. So you should be good to go with any version of go-capnproto.

**Reply by angelskieglazki** [reply]
thank you Kenton 🤗
